src/mono/wasm/runtime/driver.c

```diff
@@ -1467,4 +1480,16 @@ EMSCRIPTEN_KEEPALIVE int mono_wasm_f64_to_i52 (int64_t *destination, double valu
 
 	*destination = (int64_t)value;
 	return I52_ERROR_NONE;
-}
\ No newline at end of file
+}
+
+EMSCRIPTEN_KEEPALIVE int mono_wasm_dispatch_to_main_thread (int sync, void *callback, void *arg0, void *arg1) {
+#if DISABLE_THREADS
+	return 0;
+#else
+	if (sync)
+		emscripten_sync_run_in_main_runtime_thread(EM_FUNC_SIG_VII, callback, arg0, arg1);
+	else
+		emscripten_async_run_in_main_runtime_thread(EM_FUNC_SIG_VII, callback, arg0, arg1);
+	return 1;
+#endif
+}
```

Comment:
I can't explain exactly why, but I have a feeling it would be better if this went through the runtime's wrapper around the emscripten primitive (ie call `mono_threads_wasm_async_run_in_main_thread_vii` and also make `mono_threads_wasm_sync_run_in_main_thread_vii` in `mono/utils/mono-threads-wasm.h`).

Reply:
That sounds fine to me, I'll do it.